### PR TITLE
Set arity and curry juxted functions

### LINKED ISSUE #1569: juxt reports wrong arity
```js
R.juxt([R.prop("a"), R.path(["a", "b"])]).length; // => 0
```

### PR DESCRIPTION
**Pull request by asaf-romano**
Closes #1569

### CODE REVIEW

**Comment by asaf-romano** [comment]
This is somewhat subjective, but the more I thought about it the more it made sense, to me, for these two functions to work on empty list of functions even though the use case is questionable.

**Comment by davidchambers** [comment]
Please rebase to remove the merge commits. :)

File: src/converge.js [hunk 1]
```diff
@@ -33,7 +33,8 @@ var pluck = require('./pluck');
  *      R.converge(add3, [multiply, add, subtract])(1, 2); //=> 4
  */
 module.exports = _curry2(function converge(after, fns) {
-  return curryN(Math.max.apply(Math, pluck('length', fns)), function() {
+  var arity = fns.length ? Math.max.apply(Math, pluck('length', fns)) : 0;
```
Comment (davidchambers) [inline src/converge.js]: What is the purpose of the `fns.length` check?

Comment (asaf-romano) [inline src/converge.js]: Math.max returns -Infinity for zero arguments, which is why converge didn't work for an empty functions list,

**Comment by asaf-romano** [comment]
@davidchambers I had a hard time trying to get rid of those zombie merge commit (couldn't squash them, at least). I'll probably just generate the branch from scratch once the review is done (well, unless you know a particular set of magic commands :) ).

**Comment by davidchambers** [comment]
> unless you know a particular set of magic commands

I have a relatively straightforward solution:

```console
$ git checkout juxt-converge
$ git reset --hard $(git merge-base master juxt-converge)  # reset to common ancestor
$ curl --location https://github.com/ramda/ramda/pull/1571.diff | patch -p1  # apply patch
$ git add src test
$ git commit -m '...'
$ git push origin juxt-converge --force  # update pull request
```

While you're at it, I suggest:

```console
$ git checkout master
$ git pull upstream master  # update local master branch (assumes "upstream" is an alias for "git@github.com:ramda/ramda.git")
$ git checkout juxt-converge
$ git rebase master  # incorporate upstream changes into feature branch
$ git push origin juxt-converge --force
```

By following the steps above you'll avoid ever introducing a merge commit on a feature branch. :)

**Comment by CrossEye** [comment]
David _always_ seems to know the right set of magic commands!  :smile:

File: test/converge.js [hunk 2]
```diff
@@ -41,4 +41,8 @@ describe('converge', function() {
     eq(f3(R.__).length, 3);
   });
 
+  it('works with empty functions list', function() {
+    eq(R.converge(R.always(0), [])(), 0);
+  });
+
```
Comment (CrossEye) [inline test/converge.js]: I wonder if we want to be a little more explicit here:

```js
var fn = R.converge(function() {return arguments.lengths;}, []);
eq(fn.length, 0);
eq(fn(), 0);
```

**Comment by CrossEye** [comment]
Overall, this looks great!

**Comment by asaf-romano** [comment]
Many many thanks David!

**Comment by davidchambers** [comment]
In your commit message and in this pull request's title you have "artity" when I believe you mean "arity". Also, I think "juxed" is missing a "t". :)

File: src/converge.js (hunk 1 above)
Comment (davidchambers) [inline src/converge.js]: I'd like to use `reduce(..., 0, fns)` here to avoid a special case.

**Comment by asaf-romano** [comment]
artity, yes, oops. but juxed was already fixed :)

**Comment by davidchambers** [comment]
@asaf-romano, would you mind updating this pull request's title to match the updated commit message?

File: src/converge.js (hunk 1 above)
Comment (davidchambers) [inline src/converge.js]: Thoughts on using `reduce` here, @asaf-romano? We do so in #1572.

**Comment by CrossEye** [comment]
Thanks for your patience.

File: src/converge.js (hunk 1 above)
Comment (davidchambers) [inline src/converge.js]: #1587

**Pull request merged**